backend/Application/Aggregates/Contract/Jobs/_06_AddTokenAddress.cs

```diff
@@ -50,18 +51,20 @@ await Policies.GetTransientPolicy(GetUniqueIdentifier(), _logger, _options.Retry
             .ExecuteAsync(async () =>
             {
                 await using var context = await _contextFactory.CreateDbContextAsync(token);
+                var connection = context.Database.GetDbConnection();
                 var tokens = await context.Tokens
-                    .Where(t => (int)t.ContractIndex == identifier)
+                    .AsNoTracking()
+                    .Where(t => (int)t.ContractIndex == identifier && t.TokenAddress == null)
                     .ToListAsync(token);
-                foreach (var cisToken in tokens.Where(cisToken => cisToken.TokenAddress == null))
+                
+                foreach (var cisToken in tokens)
```

Comment:
Since this is iterating synchronously, won't that mean the entire list of tokens will need to be loaded into memory before it can actually start iterating?
There seems to be an async enumerable, which I assume will stream the rows https://learn.microsoft.com/en-us/dotnet/api/system.data.entity.dbset?view=entity-framework-6.2.0#explicit-interface-implementations

Reply:
Tried it out but can’t then reuse the database connection when doing the update, since the async call is still active.

Even though there is a lot of tokens I think it is ok once to do the in memory load.